File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque

def dfs(k, graph, visited):
    visited[k] = 1 
    for i in range(len(graph[k])):
        if visited[i] == 0 and graph[k][i] == 1:
            dfs(i, graph, visited)
    

def solution(n, computers):
    answer = 0 
    visited = [0] * n
    
    for i in range(n):
        if visited[i] == 0:
            dfs(i, computers, visited)
            answer += 1
    
        
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    
    static boolean[] visited;
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        visited = new boolean[n];
        
        for (int i = 0; i < n; i++) {
            if (!visited[i]) {
                answer++;
                dfs(i, computers);
            }
        }
        return answer;
    }
    
    static void dfs(int x, int[][] computers) {
        visited[x] = true;
        
        for (int j = 0; j < computers[x].length; j++) {
            if (x == j || visited[j]) continue;
            
            if (computers[x][j] == 1) dfs(j, computers);
        }
    }
}